privacy_consent/models/mail_mail.py

```diff
@@ -11,26 +11,27 @@ def _postprocess_sent_message(self, success_pids, failure_reason=False,
                                   failure_type=None):
         """Write consent status after sending message."""
         # Know if mail was successfully sent to a privacy consent
-        if (
-            self.mail_message_id.model == "privacy.consent"
-            and self.state == "sent"
-            and success_pids
-            and not failure_reason
-            and not failure_type
-        ):
-            # Get related consent
-            consent = self.env["privacy.consent"].browse(
-                self.mail_message_id.res_id,
-                self._prefetch,
-            )
-            # Set as sent if needed
+        for mail in self:
             if (
-                consent.state == "draft"
-                and consent.partner_id.id in {par.id for par in success_pids}
+                mail.mail_message_id.model == "privacy.consent"
+                and mail.state == "sent"
+                and success_pids
+                and not failure_reason
+                and not failure_type
             ):
-                consent.write({
-                    "state": "sent",
-                })
+                # Get related consent
+                consent = self.env["privacy.consent"].browse(
+                    mail.mail_message_id.res_id,
+                    mail._prefetch,
+                )
```

Comment:
Store the value of `mail.mail_message_id.res_id` in a list, so you can search them all together outside the loop. This way you just issue 1 `SELECT` per process instead of 1 per record.

Reply:
done